pvlib/iam.py

```diff
@@ -20,7 +20,7 @@
     'physical': {'n', 'K', 'L'},
     'martin_ruiz': {'a_r'},
     'sapm': {'B0', 'B1', 'B2', 'B3', 'B4', 'B5'},
-    'interp': set()
+    'interp': {'theta_ref', 'iam_ref'}
```

Comment:
Right now including `method` and `normalize` keys (which are optional parameters for `iam.interp`) in the `module_parameters` dict has no effect.  Should they be included here?  Off the top of my head, I'm not familiar enough with PVSystem/ModelChain's model inference code to know if that's a good idea or not.

Reply:
I think it okay to omit `method` and `normalize`. The inference functions look for the minimal set of parameters for a model, rather than a complete set (required and optional).

Reply:
It would certainly be inconvenient to the user if the inference required `method` and `normalized` to be specified as well.  I am just noting that in this PR's current state, I don't see a way to pass `method='cubic'` to `iam.interp` using ModelChain.

Reply:
That caught my attention since all models go through that. Mind if I work it out in another issue & PR?

Reply:
I think a primary use case for `interp` is to use IAM profiles defined in PAN files, which (to my knowledge) are intended to be used with cubic interpolation.  I can imagine someone specifying `method='cubic'` and not realizing that ModelChain silently ignored it and used linear interpolation anyway.  

So I'm a bit hesitant to merge this PR as long as that is a problem.  If we are sure to get to it before the next release, then OK.  But I think it would be cleaner to address it in this PR if we can.

Reply:
Done! Added a test for those kwargs, only for `physical` and `interp` iam models since they are the ones with extra optional parameters. Should a notice be added in the changelog along the following lines?

```
Bug fixes
~~~~~~~~~
* :py:class:`pvlib.pvsystem.PVSystem` now correctly passes ``n_ar`` module
   parameter to :py:func:`pvlib.iam.physical` when this IAM model is specified
   or inferred. (:ghpull:`1832`)
```